ksqldb-common/src/main/java/io/confluent/ksql/function/UdfIndex.java

```diff
@@ -366,33 +363,17 @@ boolean accepts(final SqlArgument argument, final Map<GenericType, SqlType> rese
       }
 
       if (GenericsUtil.hasGenerics(type)) {
-        return reserveGenerics(type, argument, reservedGenerics);
-      }
-
-      return ParamTypes.areCompatible(argument, type, allowCasts);
-    }
-    // CHECKSTYLE_RULES.ON: BooleanExpressionComplexity
-
-    private static boolean reserveGenerics(
-        final ParamType schema,
-        final SqlArgument argument,
-        final Map<GenericType, SqlType> reservedGenerics
-    ) {
-      if (!GenericsUtil.instanceOf(schema, argument)) {
-        return false;
-      }
-      final Map<GenericType, SqlType> genericMapping =
-          GenericsUtil.resolveGenerics(schema, argument);
-
-      for (final Entry<GenericType, SqlType> entry : genericMapping.entrySet()) {
-        final SqlType old = reservedGenerics.putIfAbsent(entry.getKey(), entry.getValue());
-        if (old != null && !old.equals(entry.getValue())) {
+        try {
+          GenericsUtil.reserveGenerics(type, argument, reservedGenerics);
+          return true;
+        } catch (final Exception e) {
```

Comment:
I guess the other thing here is if we end up swallowing exceptions we don't want to, which actually seems pretty possible. In the old way, we had `udfUtil.resolveGenerics` -> `resolveGenerics(2)` -> `resolveGenerics(3)`. `resolveGenerics(2)` threw errors depending on what happened in `resolveGenerics(3)` and we didn't catch those anywhere, right? So in theory those should still be bubbling up through the new `genericsUtil.reserveGenerics` but in this case we'd catch those too. Sorry for going back and forth, but now it seems like we should change `reserveGenerics` to return a `bool` and call another helper function in `genericsUtil` to throw an error, or just keep both `reserveGenerics` with new names. It'd be helpful to see what some other people think though. cc: @agavra @guozhangwang